payjoin/src/receive/v2/session.rs

```diff
@@ -82,20 +82,10 @@ impl SessionHistory {
     /// Receiver session Payjoin URI
     pub fn pj_uri<'a>(&self) -> Option<PjUri<'a>> {
         self.events.iter().find_map(|event| match event {
-            SessionEvent::Created(session_context) => {
-                // TODO this code was copied from ReceiverInitialized::pj_uri. Should be deduped
-                use crate::uri::{PayjoinExtras, UrlExt};
-                let id = session_context.id();
-                let mut pj = subdir(&session_context.directory, &id).clone();
-                pj.set_receiver_pubkey(session_context.s.public_key().clone());
-                pj.set_ohttp(session_context.ohttp_keys.clone());
-                pj.set_exp(session_context.expiry);
-                let extras = PayjoinExtras {
-                    endpoint: pj,
-                    output_substitution: OutputSubstitution::Disabled,
-                };
-                Some(bitcoin_uri::Uri::with_extras(session_context.address.clone(), extras))
-            }
+            SessionEvent::Created(session_context) => Some(crate::receive::v2::create_pj_uri(
+                session_context,
+                OutputSubstitution::Disabled,
```

Comment:
Thanks for pointing out the mismatch. Output substitution defaults to enabled, unless the Payjoin version is v1. We record the version in the `UncheckedProposal` session event. So the correct behavior should be: if the session log contains `UncheckedProposal`, we inspect its version and disable output substitution if it's v1, otherwise leave it enabled. If `UncheckedProposal` is missing, we default to enabling output substitution.

Reply:
This can be done in a follow up or included in this PR. I'm inclined to just merge this as is.